### PR TITLE
LG-3314: Prevents disabled buttons from submitting forms

### PR DESCRIPTION
**Pull request by bruugey**
## ✍️ Proposed changes

🎟 _Jira ticket:_ [LG-3314:](https://jira.mongodb.org/browse/LG-3314:)

### For bug fixes, new features & breaking changes

- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

### For new components

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README
- [ ] I have verified the Live Example looks as intended on the design website.

### CODE REVIEW

**Comment by spark33** [comment]
Have we actually tested this inside a form element? I thought `onSubmit` handlers were put on the form element, not the submit button, and called from there?

**Comment by bruugey** [comment]
The demo given to me was: https://codesandbox.io/s/eager-water-vrjrfz?file=/src/App.tsx

**Comment by bruugey** [comment]
> Have we actually tested this inside a form element? I thought `onSubmit` handlers were put on the form element, not the submit button, and called from there?

I don't think it's in this component's scope to read from an associated form (although with contextually-aware components we could make that a reality one day) for now I just wanted to solve what was in scope for the ticket. See JIRA for details.

File: .changeset/few-melons-refuse.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@leafygreen-ui/button': minor
+---
+
+Prevents `onSubmit` handler from firing when `disabled="true"` or `isLoading="true"`
```
Comment (spark33) [inline .changeset/few-melons-refuse.md:5]: We should update this changelog comment to be more reflective of the new change

**Review by spark33 — APPROVED** [review]

File: .changeset/few-melons-refuse.md (hunk 1 above)
Comment (bruugey) [reply .changeset/few-melons-refuse.md]: Updated, lmk what you think.

**Pull request merged**